apps/teams-test-app/src/components/NestedAppAuthAPIs.tsx

```diff
@@ -12,9 +30,144 @@ const NestedAppAuthAPIs = (): ReactElement => {
       onClick: async () => `NAA channel ${nestedAppAuth.isNAAChannelRecommended() ? 'is' : 'is not'} recommended`,
     });
 
+  const SendMessageToNestedAppAuthBridge = (): React.ReactElement =>
+    ApiWithTextInput({
+      name: 'SendMessageToNestedAppAuthBridge',
+      title: 'Send NAA Message to NestedAppAuth Bridge',
+      onClick: {
+        validateInput: () => {},
+        submit: async (input, setResult) => {
+          const bridge = (window as Window & { nestedAppAuthBridge?: NestedAppAuthBridge }).nestedAppAuthBridge;
+          if (!bridge) {
+            setResult('Bridge not available');
+            return 'Bridge not available';
+          }
+
+          // Define listener for responses
+          const listener = (response: unknown): void => {
+            setResult(JSON.stringify(response, null, 2));
+            bridge.removeEventListener?.('message', listener);
+          };
+
+          // Add event listener
+          bridge.addEventListener?.('message', listener);
+          bridge.postMessage?.(JSON.stringify(input));
```

Comment:
If this is really only for NAA messages you may want to validate that input is a valid NAA message. Right now the default input is an NAA message but it seems like this function will send pretty much any string to the bridge.

Reply:
(You can do this by filling out the validateInput function on line 38)

Reply:
Added. Thank you